### PR TITLE
feat: add admin journey to provider

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### <samp>🤖 Generated by Copilot at b0e2ab5</samp>

Refactored the admin app to use a custom hook for fetching journey data with additional fields. Added a `UserTeamRole` enum and an `ADMIN_JOURNEY_FIELDS` fragment to the journeys UI library. Updated the `JourneyProvider` component to accept both journey and admin journey types. Added tests for the new hook and fragment. Fixed some import paths in the reports pages.

- Link to Basecamp Todo

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

### <samp>🤖 Generated by Copilot at b0e2ab5</samp>

*  Extract `useAdminJourneyQuery` hook and `GET_ADMIN_JOURNEY` query from `JourneyEditPage` component to a separate folder in `libs/journeys-admin/src` ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L1-R1), [link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L14), [link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L28-R32), [link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-3873b898541aff5b78e0bc4aba59192c3d0e767d86c2e57185f5c88dd3f02063L51-R46), [link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-285d4b42ece69b54a331117869cf15038c2eff1a424ed0776e786bf072b86f68R1), [link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-c3e28a936556bf8118a1e9bdeb8bf3cf3841774ac13902659d6e1c2d8bb601d6R1-R31))
* Add `useAdminJourneyQuery.spec.tsx` file to test the `useAdminJourneyQuery` hook with a mocked provider and a sample journey data ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-544fbe3d616358970e842c16fa1b0d4b6b6bfca34a12376b059ed1089d6d67d2R1-R106))
* Import `useAdminJourneyQuery` hook and `GET_ADMIN_JOURNEY` query from the new folder in `JourneyReportsPage` and `JourneyReportsVisitorsPage` components ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-fd55f0b59a40cc07f4ea8ad5b314119563477d59a19267ed74705d84595e668aL22-R23), [link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L30-R31))
* Add `UserTeamRole` enum to `globalTypes.ts` file to define the possible roles of a user in a team ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518L6-R6), [link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-98c806310428fde902045482ce23027cd6ab7578d9e4048de778be28eed4b4caR154-R158))
* Add `email` field to the `GetAdminJourney_journey_userJourneys_user` interface to display the user's email in the admin app ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518R598))
* Add `tags` field to the `GetAdminJourney_journey` interface to display the tags assigned to the journey in the admin app ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518R694))
* Add interfaces for the team members, roles, and tags associated with the journey to the `GetAdminJourney` query ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-9fecf255f6a72895588618cbc843b8a568d1f488e250d433135d9762c70dc518L628-R670))
* Define `ADMIN_JOURNEY_FIELDS` fragment in `adminJourneyFields.ts` file, which extends the `JOURNEY_FIELDS` fragment and adds the fields for the user journeys, team, and tags ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-b636f12c48f0c023a6c5e524ef631dc54fdb933eb09d68e1cef1615a3cb439f4R1-R50))
* Import `AdminJourneyFields` fragment from `./__generated__/AdminJourneyFields` and change the `journey` type in the `Context` interface to `Journey | AdminJourney` in the `JourneyProvider` component ([link](https://github.com/JesusFilm/core/pull/1944/files?diff=unified&w=0#diff-6d170b5306d676c87a9d5eb441355ad55988db7dd9d7b1e22964a5fa19d35121L3-R7))

### CODE REVIEW

**Review by Kneesal — APPROVED** [review]
love it

**Review by tataihono — CHANGES_REQUESTED** [review]
can you use journeyAdmin as the query name instead of adminJourney so that the files for these all show up in similar alphabetical order.

File: apps/journeys-admin/src/libs/journeyAdminExists/journeyAdminExists.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const GET_ADMIN_JOURNEY = gql`
+  query GetAdminJourney($adminJourneyId: ID!) {
+    adminJourney(id: $adminJourneyId, idType: databaseId) {
+      id
+      template
+    }
+  }
+`
+
+export async function journeyAdminExists(
```
Comment (csiyang) [inline apps/journeys-admin/src/libs/journeyAdminExists/journeyAdminExists.ts:17]: journey check

File: apps/journeys-admin/src/libs/useJourneyAdminQuery/useJourneyAdminQuery.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  ${JOURNEY_ADMIN_FIELDS}
+  query GetJourneyAdmin($id: ID!) {
+    journey: adminJourney(id: $id, idType: databaseId) {
+      ...JourneyAdminFields
+    }
+  }
+`
+
+// For auth required pages in admin app
+export function useJourneyAdminQuery(
```
Comment (csiyang) [inline apps/journeys-admin/src/libs/useJourneyAdminQuery/useJourneyAdminQuery.ts:20]: journeyAdmin query

File: libs/journeys/ui/src/libs/JourneyProvider/JourneyProvider.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import { ReactElement, ReactNode, createContext, useContext } from 'react'
 
+import { JourneyAdminFields as JourneyAdmin } from './__generated__/JourneyAdminFields'
 import { JourneyFields as Journey } from './__generated__/JourneyFields'
 
 interface Context {
-  journey?: Journey
+  journey?: Journey | JourneyAdmin
```
Comment (csiyang) [inline libs/journeys/ui/src/libs/JourneyProvider/JourneyProvider.tsx:7]: provider update

File: libs/journeys/ui/src/libs/JourneyProvider/journeyAdminFields.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { gql } from '@apollo/client'
+
+import { JOURNEY_FIELDS } from './journeyFields'
+
+/**
+ * Inherit common journey fields.
+ * Add project-specific fields for reuse in journey query hooks.
+ */
+export const JOURNEY_ADMIN_FIELDS = gql`
```
Comment (csiyang) [inline libs/journeys/ui/src/libs/JourneyProvider/journeyAdminFields.ts:9]: admin fields

**Review by csiyang — COMMENTED** [review]
Comment on the relevant changes, rest is just updated types

**Review by tataihono — CHANGES_REQUESTED** [review]

**Comment by tataihono** [comment]
no longer required.

**Pull request closed without merging**